src/lib/fetcher/fetcher.ts

```diff
@@ -52,19 +60,29 @@ export default class Fetcher<T> {
       return;
     }
 
-    const promises = this.requests.map<Promise<Error | Response<T>>>(
+    const promises = this.requests.map<Promise<Error | Response<T> | SkippingRequest>>(
       async (reqInfo: RequestInfoWithHandler<T>) => {
         const info = reqInfo.info;
         let options = info.requestOptions;
+
         if (info.authenticate) {
           info.authenticate(options);
         }
 
         try {
-          const result = await this.requestWrapper.request({
-            timeout: FETCH_TIMEOUT_MS,
-            ...options,
-          });
+          let result;
+          if (info.requestFunc && isFunction(info.requestFunc)) {
```

Comment:
isFunction is already guaranteed by typing construct

Reply:
@mwamedacen updated